pravega/pravega
# Issue 6999: Add default Retention Policy

**Change log description**  
The PR aims to add a flag for setting global retention policy per stream which creation. PR also adds the logic to read and create the retention policy by reading the config param values passed.

**Purpose of the change**  
Fixes #6999 

**What the code does**  
The code will read the config param for global retention policy flag value, if it's set to true then during the stream creation it will create the retention policy and will update the StreamConfiguration object with updated RetentionPolicy object before the stream creation.

**How to verify it**  
Test should be there to check all scenarios:-
Case1- Flag=true, Min value = provided, Max value = provided, Type = time
Case2- Flag=true, Min value = provided, Max value = not provide, Type = time
Case3- Flag=true, Min value = provided, Max value = provided, Type = size
Case4- Flag=true, Min value = provided, Max value = not provided, Type = size
Case5- Flag=false, Min value = not provided, Max value = not provided, Type = empty
